tests_e2e/scripts/execute_tests_container.sh

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+
+set -euxo pipefail
+
+az login --service-principal --username "$AZURE_CLIENT_ID" --password "$AZURE_CLIENT_SECRET" --tenant "$AZURE_TENANT_ID"
+
+az acr login --name waagenttests > /dev/null
+
+docker pull waagenttests.azurecr.io/waagenttests:latest
+
+# Logs will be placed in this location. Make waagent (UID 1000 in the container) the owner.
+mkdir "$HOME/logs"
+sudo chown 1000 "$HOME/logs"
+
+docker run --rm \
+      --volume "$BUILD_SOURCESDIRECTORY:/home/waagent/WALinuxAgent" \
```

Comment:
"--volume <host_path>:<container_path>" shares the local <host_path> with the container as <container-path>.

we are sharing the agent's code and the the ssh key to connect to the test VMs (both of which Azure Pipelines downloaded to the VM); also, we are specifying a location where the container will drop the logs and then the host VM will pick them up from there to report results.